model/inline/src/main/scala/aqua/model/inline/ArrowInliner.scala

```diff
@@ -76,51 +76,66 @@ object ArrowInliner extends Logging {
   private def inline[S: Mangler: Arrows: Exports](
     fn: FuncArrow,
     call: CallModel
-  ): State[S, (OpModel.Tree, List[ValueModel])] =
-    getOutsideStreamNames.flatMap { outsideDeclaredStreams =>
-      // Function's internal variables will not be available outside, hence the scope
-      Exports[S].scope(
-        for {
-          // Process renamings, prepare environment
-          tr <- prelude[S](fn, call)
-          (tree, results) = tr
-
-          // Register captured values as available exports
-          _ <- Exports[S].resolved(fn.capturedValues)
-          _ <- Mangler[S].forbid(fn.capturedValues.keySet)
-
-          // Now, substitute the arrows that were received as function arguments
-          // Use the new op tree (args are replaced with values, names are unique & safe)
-          callableFuncBodyNoTopology <- TagInliner.handleTree(tree, fn.funcName)
-          callableFuncBody =
-            fn.capturedTopology
-              .fold[OpModel](SeqModel)(ApplyTopologyModel.apply)
-              .wrap(callableFuncBodyNoTopology)
-
-          opsAndRets <- pushStreamResults(
-            outsideDeclaredStreams,
-            call.exportTo,
-            results,
-            callableFuncBody
-          )
-          (ops, rets) = opsAndRets
-        } yield SeqModel.wrap(ops.reverse: _*) -> rets.reverse
-      )
+  ): State[S, (SeqModel.Tree, List[ValueModel], Map[String, ValueModel], Map[String, FuncArrow])] =
```

Comment:
Please, add comment on what this function returns. Also I think it would be better to define a `case class` for return type at this point.